### PR TITLE
fix: iteration-item scope and raise error when evaluated value is unexpected type in ControlActivities

### LINKED ISSUE #102: ForEach item() is not available to child activities of child activities
It seems like the foreach item() should be available to child activities of child activities. Investigate and introduce fix for scoping mechanism.

### PR DESCRIPTION
**Pull request by arjendev**
Fixes #102 and #103

### CODE REVIEW

File: src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py
Comment (LeonardHd) [inline src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py:1]: I suggest we also update the __init__.py

File: src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class ControlActivityExpressionEvaluatedNotToExpectedTypeError(UserError):
+    """ControlActivityExpressionEvaluatedNotToExpectedType.
+
+    This exception is raised when a ControlActivities iteration expression is not evaluating to the expected type.
+    This might be due incorrect expression or incorrectly registering activity results (e.g. registering a dictionary instead of expected list)
+    """
+
+    def __init__(self, activity_name: str, expected_type: str) -> None:
```
Comment (LeonardHd) [inline src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py:11]: Suggestion: to use the actual Type type and then convert to a str representation.

**Review by ydaponte — DISMISSED** [review]
All working now! Thanks so much!

File: src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py (hunk 1 above)
Comment (arjendev) [reply src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py:11]: Good suggestion, addressed!

File: src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py
Comment (arjendev) [reply src/data_factory_testing_framework/exceptions/_control_activity_expression_evaluated_not_to_expected_type.py:1]: Nice catch, addressed!

**Review by LeonardHd — APPROVED** [review]

**Review by ydaponte — APPROVED** [review]
LGTM!

**Pull request merged**